### ISSUE TITLE
Docker not mounting directory properly

### ISSUE THREAD

**Issue by workmonitored** [docker/for-win#472]
### Expected behavior
Docker should mount the volume correctly during initial start. 
### Actual behavior
Docker does not mount the volume correctly. The files are not visible during initial starting of windows 10 system. Stopping and starting container is needed. 
### Information

  - Diagnostic ID from "Diagnose & Feedback" in the menu.
495E0286-6973-4DD1-8ABD-83199CE63B9C/2017-02-07_13-04-16
  - a reproducible case if this is a bug, Dockerfiles FTW
version: '2'                                                             
services:                                                                
  gogs:                                                                  
   image: gogs/gogs                                                      
   restart: always                                                       
   networks:                                                             
       common-net:                                                       
           aliases:                                                      
              - gogs                                                     
   ports:                                                                
   - "10080:3000"                                                        
   - "10022:22"                                                          
   volumes:                                                              
   - /c/web/gdrive/docker/data/gogs:/data                                
                                                                         
networks:                                                                
  common-net:                                                            

  - page URL if this is a docs issue or the name of a man page
  - host distribution and version (Windows version, build number, etc)
    Windows 10 - Enterprise (version 1511) -  OS.Build 10586.318

**Comment by simonferquel** [comment]
Thanks for the report. It is a tricky issue as for now, we need the Docker daemon to be running to mount shared drives (and when starting, the daemon automatically start the containers marked as "restart: always").

@justincormack could we do something in moby (or even just in our mount/nsenter script) to make the cifs mount persistent across reboot to make sure it is mounted before starting dockerd ?

**Comment by justincormack** [comment]
@simonferquel sure, you just need to pass the credentials in.

**Comment by rneugeba** [comment]
I'm closing this as a duplicate of https://github.com/docker/for-win/issues/71. Please continue the discussion there. @workmonitored I'll ping you on that issue, so that you get updates from there
